sksurv/util.py

```diff
@@ -260,3 +260,83 @@ def safe_concat(objs, *args, **kwargs):
         concatenated[name] = pd.Categorical(concatenated[name], **params)
 
     return concatenated
+
+
+class _ConditionalAvailableProperty:
+    """Implements a conditional property using the descriptor protocol based on the property decorator.
+
+    The corresponding class in scikit-learn (_AvailableIfDescriptor) only supports callables. This
+    class adopts the property decorator as presented by the descriptor guide in the offical documentation.
+
+    The check is defined on the getter function. Setter, Deleter also need to be defined on the exposing class
+    if they are supposed to be available.
+
+    See Also
+    --------
+    sklearn.utils.available_if._AvailableIfDescriptor:
+    The original class in scikit-learn.
+    """
+
+    def __init__(self, check, fget=None, fset=None, fdel=None, doc=None):
+        self._check = check
+        self.fget = fget
+        self.fset = fset
+        self.fdel = fdel
+        if doc is None and fget is not None:
+            doc = fget.__doc__
+        self.__doc__ = doc
+        self._name = ''
+
+    def _run_check(self, obj):
+        attr_err = AttributeError(
+            f"This {repr(obj)} has no attribute {repr(self._name)}"
```

Comment:
Instead of `{repr(obj)}`, you can use `{obj!r}`, and move the definition of `attr_err` into the if block.